<launch>
  <arg name="usb_port"                default="/dev/ttyUSB0"/>
  <arg name="dxl_baud_rate"           default="57600"/>
  <arg name="namespace"               default="dynamixel_workbench"/>

  <param name="dynamixel_info"          value="$(find uav_ros_control)/launch/wheel_mode.yaml"/>

  <node name="$(arg namespace)" pkg="dynamixel_workbench_controllers" type="dynamixel_workbench_controllers"
        required="true" output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
    <rosparam>
      publish_period: 0.010
      dxl_read_period: 0.010
      dxl_write_period: 0.010
      use_moveit: false
      use_joint_states_topic: true
      use_cmd_vel_topic: true
      mobile_robot_config:                <!--this values will be set when 'use_cmd_vel_topic' is true-->
        seperation_between_wheels: 0.160  <!--default value is set by reference of TB3-->
        radius_of_wheel: 0.033            <!--default value is set by reference of TB3-->
    </rosparam>
  </node>

  <node name="dynamixel_net" pkg="uav_ros_control" type="dynamixel_net.py" output="screen">
    <param name="rate" value="50" />
    <param name="fully_winded" value="0" />
    <param name="fully_unwinded" value="10" />
    <param name="winding_speed" value="1.5" />
  </node>
</launch>
